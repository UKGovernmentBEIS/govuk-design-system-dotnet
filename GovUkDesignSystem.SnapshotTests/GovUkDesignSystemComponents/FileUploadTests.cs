using System.Collections.Generic;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class FileUploadTests : SnapshotTestBase
    {
        private FileUploadViewModel DefaultFileUploadViewModel()
        {
            return new FileUploadViewModel
            {
                Id = "test-id",
                Name = "test-name",
                Attributes = new Dictionary<string, string> { { "attr-name", "attr-value" } },
                Classes = "cssClass",
                DescribedBy = "test-described-by",
                Label = new LabelViewModel { Text = "test-label" },
                Hint = new HintViewModel { Text = "test-hint" },
                ErrorMessage = new ErrorMessageViewModel { Text = "test-error" },
                FormGroup = new FormGroupViewModel { Classes = "form-group-classes" }
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("FileUpload", DefaultFileUploadViewModel());
        }

        [Fact]
        public async Task Render_NoLabel()
        {
            // Arrange
            var viewModel = DefaultFileUploadViewModel();
            viewModel.Label = null;

            // Act & Assert
            await VerifyPartial("FileUpload", viewModel);
        }

        [Fact]
        public async Task Render_NoHint()
        {
            // Arrange
            var viewModel = DefaultFileUploadViewModel();
            viewModel.Hint = null;

            // Act & Assert
            await VerifyPartial("FileUpload", viewModel);
        }

        [Fact]
        public async Task Render_NoError()
        {
            // Arrange
            var viewModel = DefaultFileUploadViewModel();
            viewModel.ErrorMessage = null;

            // Act & Assert
            await VerifyPartial("FileUpload", viewModel);
        }

        [Fact]
        public async Task Render_NoDescribedBy()
        {
            // Arrange
            var viewModel = DefaultFileUploadViewModel();
            viewModel.DescribedBy = null;
            viewModel.ErrorMessage = null;
            viewModel.Hint = null;

            // Act & Assert
            await VerifyPartial("FileUpload", viewModel);
        }
    }
}
