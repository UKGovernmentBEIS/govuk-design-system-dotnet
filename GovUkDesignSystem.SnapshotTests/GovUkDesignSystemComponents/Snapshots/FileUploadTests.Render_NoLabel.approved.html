
<div class="govuk-form-group form-group-classes govuk-form-group--error">

<div class="govuk-hint"
      id="test-id-hint"
      >

test-hint</div>
<span class="govuk-error-message"
      id="test-id-error"
      >
    <span class="govuk-visually-hidden">Error</span>

test-error</span>    <input class="govuk-file-upload  cssClass govuk-file-upload--error"
           id=test-id
           name=test-name
           type="file"
           aria-describedby="test-described-by test-id-hint test-id-error"
           attr-name="attr-value">
</div>
